examples/mpm_lagrangian_forces.py

```diff
@@ -90,23 +82,23 @@ def grid_op():
         if grid_m[i, j] > 0:
             inv_m = 1 / grid_m[i, j]
             grid_v[i, j] = inv_m * grid_v[i, j]
-            grid_v(1)[i, j] -= dt * 9.8
+            grid_v[i, j].y -= dt * 9.8
 
-            # center sticky circle
+            # center collision circle
             dist = ti.Vector([i * dx - 0.5, j * dx - 0.5])
             if dist.norm_sqr() < 0.005:
                 dist = dist.normalized()
-                grid_v[i, j] -= dist * grid_v[i, j].dot(dist)
+                grid_v[i, j] -= dist * min(0, grid_v[i, j].dot(dist))
 
             # box
-            if i < bound and grid_v(0)[i, j] < 0:
-                grid_v(0)[i, j] = 0
-            if i > n_grid - bound and grid_v(0)[i, j] > 0:
-                grid_v(0)[i, j] = 0
-            if j < bound and grid_v(1)[i, j] < 0:
-                grid_v(1)[i, j] = 0
-            if j > n_grid - bound and grid_v(1)[i, j] > 0:
-                grid_v(1)[i, j] = 0
+            if i < bound and grid_v[i, j].x < 0:
+                grid_v[i, j].x = 0
+            if i > n_grid - bound and grid_v[i, j].x > 0:
+                grid_v[i, j].x = 0
+            if j < bound and grid_v[i, j].y < 0:
+                grid_v[i, j].y = 0
+            if j > n_grid - bound and grid_v[i, j].y > 0:
+                grid_v[i, j].y = 0
```

Comment:
Cool! Will we adapt this usage in other examples, like mpm128?

Reply:
Sounds good! But let me focus this PR on a single file.